libraries/core-react/src/utils/test-utils.tsx

```diff
@@ -1,17 +1,21 @@
 import * as React from 'react'
-import { ElementType, ReactElement, ReactNode } from 'react'
+import { ElementType, ReactElement, ReactNode, MutableRefObject } from 'react'
 import { render, RenderOptions, RenderResult } from '@testing-library/react'
 
 const AllTheProviders = ({ children }: { children: ReactNode }) => {
-  const anchorRef = React.useRef()
+  const anchorRef = React.useRef(null)
   const [anchorEl, setAnchorEl] = React.useState<ElementType>(null)
 
   React.useEffect(() => {
-    setAnchorEl(anchorRef.current)
+    if (anchorRef.current) {
+      setAnchorEl(anchorRef.current)
+    }
   }, [anchorRef])
 
   const updatedChildren = React.Children.map(children, (child) =>
-    React.cloneElement(child as ReactElement, { anchorEl }),
+    React.cloneElement(child as ReactElement, {
+      anchorEl,
+    }),
   ) as ReactNode
 
   return (
```

Comment:
Good job on making this a shared testing util 👍

I suggest we some more changes, as its not very evident in testing files what `test-utils` does...

We can make a `test` folder under `utils` and add a file called for example `withAnchorEl`. 

This way its more explicit when using the custom render in test files such as;
```typescript
import { render, cleanup, screen, fireEvent } from '@utils/test/withAnchorEl'
```
instead of;
```typescript
import { render, cleanup, screen, fireEvent } from '../../utils/test-utils'
```